NangoHQ/nango
# fix(connect): working e2e, use default values from session

## Describe your changes

Fixes https://linear.app/nango/issue/NAN-1836/use-default-values-from-sessiontoken
Fixes https://linear.app/nango/issue/NAN-1835/filters-get-integrations-with-session-token

> [!NOTE]
> In this PR, the whole UI should work without additional setup, but can only accessed with a connect token through the dashboard

- Filter `GET /integrations` when using session token
- Use default values in the UI if filled
It's a bit tricky because I assumed I auto-triggered the form only when there was no field, but if you pre-fill all values then it should also auto-trigger. But now if you badly prefill a field it's hard to show (I assumed prefilled values should not be shown in the UI). 
I also need all fields to be in the DOM even when hidden.

- Rollback auth for GET /integrations/{id}
I wrongly stated that I needed that, I don't.

- Fixed nango not being properly reloaded because it was a ref


## 🧪 Test

- Go to `packages/server/lib/controllers/v1/connect/sessions/postConnectSessions.ts`
- Add some integrations filtering and default values
- Go to the dashboard and trigger the Connect UI

---

> Example of badly preconfigured fields, ideally it should not happen but we have a way to warm the user

<img width="498" alt="Screenshot 2024-10-02 at 16 20 33" src="https://github.com/user-attachments/assets/e285efd0-6e10-4ce6-a7a2-6af1ff9be134">
